### DISCUSSION TITLE
Custom input validations in login, password reset pages

### DISCUSSION THREAD

**Discussion by chathu03j** [keycloakify/keycloakify#255] [Q&A]
I want to add some custom validations such as input required, email format validation, etc.. in the login and password reset pages. I know we can use custom validations on user inputs in pages such as user registration, but I'm not sure how we can customize the validations in login forms. Any thoughts?

**Comment by garronej** [comment]
Hi @chathu03j,  
To my knowelage it isn't possible.  
The user profile is only made available by keycloak in the `register-user-profile.ftl` page.  

If you are willing to go into the trouble of implementing your own Keycloak extention you should be able to acheive what you want.  
This not directly related answer from Keycloak suport staff might be helpfull: https://keycloak.discourse.group/t/make-password-policies-available-to-freemarker/11632  

Anyway, if you find a way, I'll be happy to get your feedback.  

Best

**Comment by chmcquee** [comment]
@chathu03j  I used formik and yup for this. Have a look at these libraries. If you have further questions I'm happy to assist.

**Reply by garronej** [reply]
Thanks for your suggestion,  
It's indeed a quick win solution if you are willing to hard code your validation params into your theme or use the realm hack to pass some extra params.  
See: https://github.com/InseeFrLab/keycloakify/issues/258#issuecomment-1456061177
